File: sdk/r/R/affiliate.R
#' Affiliate: affiliate program management
#' @param http HttpClient object
#' @return Resource object
#' @export
AffiliateResource <- function(http) {
  env <- new.env(parent = emptyenv())
  env$http <- http
  class(env) <- "AffiliateResource"
  env
}

#' GET /api/affiliates/summary - Get Affiliate summary
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getSummary <- function(self, ...) {
  self$http$get("/api/affiliates/summary", list(...))
}

#' GET /api/affiliates/campaigns - Get Affiliate campaigns
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listCampaigns <- function(self, ...) {
  self$http$get("/api/affiliates/campaigns", list(...))
}

#' GET /api/affiliates/commissions - Get Affiliate commissions
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listCommissions <- function(self, ...) {
  self$http$get("/api/affiliates/commissions", list(...))
}

#' GET /api/affiliates/payouts - Get Affiliate payouts
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listPayouts <- function(self, ...) {
  self$http$get("/api/affiliates/payouts", list(...))
}

#' GET /api/affiliates/vouchers - Get Affiliate vouchers
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listVouchers <- function(self, ...) {
  self$http$get("/api/affiliates/vouchers", list(...))
}

#' GET /api/affiliates/commissionplans - Get Affiliate commission plans
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listCommissionPlans <- function(self, ...) {
  self$http$get("/api/affiliates/commissionplans", list(...))
}

File: sdk/c/src/affiliate.c
#include "hitechcloud/affiliate.h"
#include <stdlib.h>


/* GET /api/affiliates/summary - Get Affiliate summary */
char *hc_affiliate_get_summary(hc_http_client_t *client) {
    return hc_get(client, "/api/affiliates/summary", NULL);
}

/* GET /api/affiliates/campaigns - Get Affiliate campaigns */
char *hc_affiliate_list_campaigns(hc_http_client_t *client) {
    return hc_get(client, "/api/affiliates/campaigns", NULL);
}

/* GET /api/affiliates/commissions - Get Affiliate commissions */
char *hc_affiliate_list_commissions(hc_http_client_t *client) {
    return hc_get(client, "/api/affiliates/commissions", NULL);
}

/* GET /api/affiliates/payouts - Get Affiliate payouts */
char *hc_affiliate_list_payouts(hc_http_client_t *client) {
    return hc_get(client, "/api/affiliates/payouts", NULL);
}

/* GET /api/affiliates/vouchers - Get Affiliate vouchers */
char *hc_affiliate_list_vouchers(hc_http_client_t *client) {
    return hc_get(client, "/api/affiliates/vouchers", NULL);
}

/* GET /api/affiliates/commissionplans - Get Affiliate commission plans */
char *hc_affiliate_list_commission_plans(hc_http_client_t *client) {
    return hc_get(client, "/api/affiliates/commissionplans", NULL);
}

